pkg/storage/stores/indexshipper/compactor/deletion/tenant_request_handler.go

```diff
@@ -0,0 +1,36 @@
+package deletion
+
+import (
+	"net/http"
+
+	"github.com/grafana/loki/pkg/storage/stores/indexshipper/compactor/retention"
+
+	"github.com/grafana/dskit/tenant"
+)
+
+func TenantMiddleware(limits retention.Limits, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		userID, err := tenant.TenantID(ctx)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		allLimits := limits.AllByUserID()
```

Comment:
nit: we have a convention of defining methods on limit structs themselves. This is easier to consume as it doesn't require much boilerplate when we check a limit and lets us define smaller interfaces over just the methods we need.